Error while checking build/O2/fullCI_slc9 for b57ebf1b212975976dd1a557bb83d1844133aec7 at 2026-03-13 10:42:
```
## sw/BUILD/O2-latest/log
CMake Error at dependencies/O2Dependencies.cmake:111 (message):
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/15152/b57ebf1b212975976dd1a557bb83d1844133aec7/build_O2_fullCI_slc9/pretty.html).

--- a/Framework/Core/CMakeLists.txt
+++ b/Framework/Core/CMakeLists.txt
@@ -8,7 +8,6 @@
 # In applying this license CERN does not waive the privileges and immunities
 # granted to it by virtue of its status as an Intergovernmental Organization
 # or submit itself to any jurisdiction.
-
 o2_add_library(Framework
                SOURCES src/AnalysisHelpers.cxx
                        src/AlgorithmSpec.cxx
@@ -178,7 +177,7 @@ o2_add_library(Framework
                                      RapidJSON::RapidJSON
                                      Arrow::arrow_shared
                                      ArrowDataset::arrow_dataset_shared
-                                     ArrowCompute::arrow_compute_shared
+                                     $<$<TARGET_EXISTS:ArrowCompute::arrow_compute_shared>:ArrowCompute::arrow_compute_shared>
                                      Microsoft.GSL::GSL
                                      O2::FrameworkLogger
                                      Gandiva::gandiva_shared

--- a/dependencies/O2Dependencies.cmake
+++ b/dependencies/O2Dependencies.cmake
@@ -92,31 +92,34 @@ if(NOT TARGET ArrowAcero::arrow_acero_shared)
   )
 endif()
 
-if(NOT TARGET ArrowCompute::arrow_compute_shared)
-  # ArrowCompute::arrow_compute_shared is linked for no reason to parquet
-  # so we cannot use it because we do not want to build parquet itself.
-  # For that reason at the moment we need to do the lookup by hand.
-  get_target_property(ARROW_SHARED_LOCATION Arrow::arrow_shared LOCATION)
-  get_filename_component(ARROW_SHARED_DIR ${ARROW_SHARED_LOCATION} DIRECTORY)
+string(REGEX MATCH "([0-9]+)\.*" ARROW_MAJOR "${ARROW_VERSION}")
+if(${ARROW_MAJOR} GREATER 20)
+  if(NOT TARGET ArrowCompute::arrow_compute_shared)
+    # ArrowCompute::arrow_compute_shared is linked for no reason to parquet
+    # so we cannot use it because we do not want to build parquet itself.
+    # For that reason at the moment we need to do the lookup by hand.
+    get_target_property(ARROW_SHARED_LOCATION Arrow::arrow_shared LOCATION)
+    get_filename_component(ARROW_SHARED_DIR ${ARROW_SHARED_LOCATION} DIRECTORY)
 
-  find_library(ARROW_COMPUTE_SHARED arrow_compute
-      PATHS ${ARROW_SHARED_DIR}
-      NO_DEFAULT_PATH
-  )
+    find_library(ARROW_COMPUTE_SHARED arrow_compute
+        PATHS ${ARROW_SHARED_DIR}
+        NO_DEFAULT_PATH
+    )
 
-  if(ARROW_COMPUTE_SHARED)
-    message(STATUS
-	    "Found arrow_compute_shared library at: ${ARROW_COMPUTE_SHARED}")
-  else()
-    message(FATAL_ERROR
-            "arrow_compute_shared library not found in ${ARROW_SHARED_DIR}")
+    if(ARROW_COMPUTE_SHARED)
+      message(STATUS
+              "Found arrow_compute_shared library at: ${ARROW_COMPUTE_SHARED}")
+    else()
+      message(FATAL_ERROR
+              "arrow_compute_shared library not found in ${ARROW_SHARED_DIR}")
+    endif()
+
+    # Step 3: Create a target for ArrowCompute::arrow_compute_shared
+    add_library(ArrowCompute::arrow_compute_shared SHARED IMPORTED)
+    set_target_properties(ArrowCompute::arrow_compute_shared PROPERTIES
+        IMPORTED_LOCATION ${ARROW_COMPUTE_SHARED}
+    )
   endif()
-
-  # Step 3: Create a target for ArrowCompute::arrow_compute_shared
-  add_library(ArrowCompute::arrow_compute_shared SHARED IMPORTED)
-  set_target_properties(ArrowCompute::arrow_compute_shared PROPERTIES
-      IMPORTED_LOCATION ${ARROW_COMPUTE_SHARED}
-  )
 endif()
 
 if(NOT TARGET Gandiva::gandiva_shared)

